### ISSUE TITLE
Projected Gradient Descent: overestimation

### ISSUE THREAD

**Issue by benjaminbenteke** [benjaminbenteke/EWO_Methods#8]
![change0](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/b5324016-b762-4751-871a-17185138efcb)
![change1](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/5c806656-454d-45cb-8c2f-8c0912a0148d)
![change2](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/d867cff5-be4a-479a-a9b0-887788850bde)
![change3](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/5a835e96-a267-447b-8789-8a9b4a5afb6e)

These are the plots based on our change on EWO without correction. I suspect the Projected Gradient Descent (PGD), especially the projection. However, I've not tried tried yet the former PGD with np. max(theta, 0). However, when testing with a toy example, it was working. I am trying this https://see.stanford.edu/materials/lsocoee364b/hw4sol.pdf.

**Comment by cvanbommel** [comment]
I don't think I understand the plots in general.  If 2023-10-14 was the first forecast submission date, isn't the first prediction at horizon 3 for 2023-11-04?

**Comment by cvanbommel** [comment]
Also, just discovered a post on StackOverflow regarding a mistake in the file you reference.  See https://stackoverflow.com/questions/74844094/projection-onto-unit-simplex-using-gradient-decent-in-pytorch for the post.  Maybe this is the source of the issue with the overestimate?

**Comment by benjaminbenteke** [comment]
No, I didn't use it. Below is what I am using:

def constraint_function(x):
    return np.array([np.sum(x)-1])


def project_into_feasible_set(x,constraint_function):
    result = minimize(lambda x_prime: 0.5*np.linalg.norm(x_prime - x, ord= 2)**2, x, constraints={'type': 'ineq', 'fun': constraint_function},bounds= [(0,1) for _ in range(len(x))])
    return result.x

#initial_x = np.array([-0.2, 0.4,0.3,4])
#project_into_feasible_set(initial_x,constraint_function)


Also, someone fixed the mistake in https://stackoverflow.com/questions/74844094/projection-onto-unit-simplex-using-gradient-decent-in-pytorch

**Comment by cvanbommel** [comment]
You have 'type': 'ineq' on your constraint function, it should have 'type': 'eq' so the weights sum to exactly 1.

**Comment by benjaminbenteke** [comment]
Yeah! I'll change it.

Thank you!

**Comment by benjaminbenteke** [comment]
They look good now. This is EWO without c
![change0](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/f0d2e4dc-ddf4-4801-a68a-a070fdcf6cf6)
![change1](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/571c5a74-405c-413c-b6e2-422fbcfe0e0e)
![change2](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/f7f74b7b-f2fd-43ca-b1f0-b733595ab5e7)
![change3](https://github.com/benjaminbenteke/EWO_Methods/assets/57951412/df1efa3a-d9e4-43c0-8452-df1f141e577d)
orrection.
